src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -115,33 +113,26 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 		}
 	}
 
-	err := r.Get(ctx, r.Key(), &sts)
-	if err != nil && !k8serrors.IsNotFound(err) {
-		return fmt.Errorf("error while fetching StatefulSet resource: %w", err)
+	obj, err := r.obj()
+	if err != nil {
+		return fmt.Errorf("unable to construct StatefulSet object: %w", err)
+	}
+	created, err := CreateIfNotExists(ctx, r, obj, r.logger)
+	if err != nil {
+		return fmt.Errorf("error while creating StatefulSet resource: %w", err)
```

Comment:
NIT: I'm not sure why here we add context, but in other places when error is return from `CreateIfNotExists` then context is not added.